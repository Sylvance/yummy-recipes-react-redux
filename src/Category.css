.Category {
    text-align: center;
  }
  
.Category-logo {
    animation: Category-logo-spin infinite 20s linear;
    height: 80px;
}

.Category-header {
    background-color: #222;
    height: 150px;
    padding: 20px;
    color: white;
}

.Category-title {
    font-size: 1.5em;
}

.Category-intro {
    font-size: large;
}

@keyframes Category-logo-spin {
    from { transform: rotate(0deg); }
    to { transform: rotate(360deg); }
}
