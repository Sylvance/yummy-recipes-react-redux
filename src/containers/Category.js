import React from 'react'
import PropTypes from 'prop-types'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import CategoryHeader from '../components/CategoryHeader'
import CategoryMainSection from '../components/CategoryMainSection'

import * as Actions from '../actions'

const Category = ({categories, meta, userid, actions}) => (
  <div>
    <CategoryHeader addCategory={actions.addCategory} userid={userid}/>
    <CategoryMainSection userid={userid} meta={meta} categories={categories} actions={actions} />
  </div>
)

Category.propTypes = {
  categories: PropTypes.array.isRequired,
  userid: PropTypes.number.isRequired,
  actions: PropTypes.object.isRequired,
  meta: PropTypes.object.isRequired
}

const mapStateToProps = state => ({
  categories: state.categories,
  meta: state.meta,
  userid: JSON.parse(localStorage.getItem('userid'))
})

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(Actions, dispatch)
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Category)
