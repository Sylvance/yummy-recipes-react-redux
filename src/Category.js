import React, { Component } from 'react';
import logo from './logo.svg';
import './Category.css';

class Category extends Component {
  render() {
    return (
      <div className="Category">
        <header className="Category-header">
          <img src={logo} className="Category-logo" alt="logo" />
          <h1 className="Category-title">Categories Page</h1>
        </header>
        <p className="Category-intro">
          To get started, edit <code>src/Category.js</code> and save to reload.
        </p>
        <div className="Category-intro">
          Yummy recipes provides a platform for users to keep track of their awesome 
          recipes and share with others if they so wish.
        </div>
      </div>
    );
  }
}

export default Category;
