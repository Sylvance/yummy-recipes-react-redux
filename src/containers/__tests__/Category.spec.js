// Auto-generated do not edit

/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import Category from '../Category';

describe('Category test', () => {
  it('Category should match snapshot', () => {
    const component = renderer.create(
        <div>
      </div>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
