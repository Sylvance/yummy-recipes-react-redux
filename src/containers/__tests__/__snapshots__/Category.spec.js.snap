// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category test Category should match snapshot 1`] = `<div />`;
